modules/index.js

```diff
@@ -49,14 +49,34 @@ var RouteHandler = React.createClass({
   componentDidMount () {
     var route = this.context.getRouteAtDepth(this.getRouteDepth());
     // TODO: do we really need this? surfacing in ember migration app
-    if (route && route.handler)
+    if (route && route.handler) {
       runHooks('setup', route.handler, this.context.asyncPropsState);
+      this.lastHandler = route.handler;
+    }
+  },
+
+  componentDidUpdate() {
+    var route = this.context.getRouteAtDepth(this.getRouteDepth());
```

Comment:
I really wish there was a way to retrieve the last handler without having to store it in `this` like it is now, sort of how you get `prevProps, prevState` inside the `componentDidUpdate()` hook.